UnityProject/Assets/Scripts/Items/Food/Donut.cs

```diff
@@ -0,0 +1,85 @@
+﻿using System.Collections;
+using UnityEngine;
+
+
+public class Donut : Edible
+{
+    [Tooltip("The amount of brute damage healed for a player who has a secuirty role.")]
+    public int SecurityHealBruteDamage = 5;
+    private int securityHealBruteDamage => SecurityHealBruteDamage;
+
+
+    /// <summary>
+	/// Checks if the player eating the donut has a secuirty job and heals them if they are part of security.
+    /// If the player is not part of secuirty then nothing special will be triggered.
+	/// </summary>
+	public override void Eat(PlayerScript eater, PlayerScript feeder)
+	{
+		if (CheckForJob(eater))
+		{
+            Heal(eater);
+		    Eat(eater, feeder, NutritionLevel);
+		}
+		else
+		{
+		    Eat(eater, feeder, NutritionLevel);
+		}
+	}
+
+    /// <summary>
+	/// Heals all body parts that have brute damage.
+	/// </summary>
+    private void Heal(PlayerScript player)
+    {
+        var livingHealth = player.GetComponent<LivingHealthBehaviour>();
+        foreach (BodyPartBehaviour BodyPart in livingHealth.BodyParts)
+		{
+            if(BodyPart.BruteDamage > 0)
+            {
+                BodyPart.HealDamage(securityHealBruteDamage, DamageType.Brute);
+            }
+		}
+    }
+
+    /// <summary>
+	/// Checks if the player has a secuirty job.
+	/// </summary>
+    private bool CheckForJob(PlayerScript JobHolder)
+    {
+        switch (JobHolder.mind.occupation.JobType)
+        {
+            case JobType.SECURITY_OFFICER:
+                return true;
+            case JobType.HOS:
+                return true;
+            case JobType.DETECTIVE:
+                return true;
+            case JobType.WARDEN:
+                return true;
+            default:
+                return false;
+        }
+    }
+	private void Eat(PlayerScript eater, PlayerScript feeder, int nutrition)
```

Comment:
Looks like this part wasn't altered at all. You could instead add your job check and heal in the public method and then call ``Base.Eat()``

Reply:
So do I delete the override function and move everything in Eat?

Reply:
let me inspect the base class.

Reply:
Ok I see what you did. Delete the private and call ``Base.Eat()`` in your overridden public **after** doing your heal check.

Reply:
```
public override void Eat(PlayerScript eater, PlayerScript feeder)
	{
		if (CheckForJob(eater)) 
		{
			Heal(eater);
		}
		Base.Eat(eater, feeder, NutritionLevel);
	}
```
like this?

Reply:
yep, that should work.